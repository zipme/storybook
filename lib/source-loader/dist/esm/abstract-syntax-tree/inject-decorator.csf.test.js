import fs from 'fs';
import path from 'path';
import injectDecorator from './inject-decorator';
describe('inject-decorator', function () {
  describe('positive - ts - csf', function () {
    it('includes storySource parameter in the default exported object', function () {
      var mockFilePath = './__mocks__/inject-decorator.ts.csf.txt';
      var source = fs.readFileSync(mockFilePath, 'utf-8');
      var result = injectDecorator(source, path.resolve(__dirname, mockFilePath), {
        parser: 'typescript'
      });
      expect(result.source).toMatchSnapshot();
      expect(result.source).toEqual(expect.stringContaining('export default {parameters: {"storySource":{"source":"import React from'));
    });
  });
  describe('injectStoryParameters - ts - csf', function () {
    it('includes storySource parameter in the default exported object', function () {
      var mockFilePath = './__mocks__/inject-parameters.ts.csf.txt';
      var source = fs.readFileSync(mockFilePath, 'utf-8');
      var result = injectDecorator(source, path.resolve(__dirname, mockFilePath), {
        injectStoryParameters: true,
        parser: 'typescript'
      });
      expect(result.source).toMatchSnapshot();
    });
  });
});