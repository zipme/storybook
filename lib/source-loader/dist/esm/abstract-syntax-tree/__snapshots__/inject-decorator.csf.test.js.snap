// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`inject-decorator injectStoryParameters - ts - csf includes storySource parameter in the default exported object 1`] = `
"


Basic.parameters = { storySource: { source: \\"() => (\\\\n  <Button onClick={action(\\\\\\"clicked\\\\\\")}>Hello Button</Button>\\\\n)\\" }, ...Basic.parameters };
WithParams.parameters = { storySource: { source: \\"() => <Button>WithParams</Button>\\" }, ...WithParams.parameters };
WithDocsParams.parameters = { storySource: { source: \\"() => <Button>WithDocsParams</Button>\\" }, ...WithDocsParams.parameters };
WithStorySourceParams.parameters = { storySource: { source: \\"() => <Button>WithStorySourceParams</Button>\\" }, ...WithStorySourceParams.parameters };
WithTemplate.parameters = { storySource: { source: \\"(args: Args) => <Button {...args} />\\" }, ...WithTemplate.parameters };
WithEmptyTemplate.parameters = { storySource: { source: \\"(args: Args) => <Button {...args} />\\" }, ...WithEmptyTemplate.parameters };
WithAddFunctionParameters.parameters = { storySource: { source: \\"() => null\\" }, ...WithAddFunctionParameters.parameters };"
`;

exports[`inject-decorator positive - ts - csf includes storySource parameter in the default exported object 1`] = `
"import React from \\"react\\";
import { action } from \\"@storybook/addon-actions\\";
import { Button } from \\"@storybook/react/demo\\";

export default {parameters: {\\"storySource\\":{\\"source\\":\\"import React from \\\\\\"react\\\\\\";\\\\nimport { action } from \\\\\\"@storybook/addon-actions\\\\\\";\\\\nimport { Button } from \\\\\\"@storybook/react/demo\\\\\\";\\\\n\\\\nexport default {\\\\n  title: \\\\\\"Button\\\\\\",\\\\n  excludeStories: [\\\\\\"text\\\\\\"],\\\\n  includeStories: /emoji.*/\\\\n};\\\\n\\\\nexport const text = () => (\\\\n  <Button onClick={action(\\\\\\"clicked\\\\\\")}>Hello Button</Button>\\\\n);\\\\n\\\\nexport const emoji = () => (\\\\n  <Button onClick={action(\\\\\\"clicked\\\\\\")}>\\\\n    <span role=\\\\\\"img\\\\\\" aria-label=\\\\\\"so cool\\\\\\">\\\\n      😀 😎 👍 💯\\\\n    </span>\\\\n  </Button>\\\\n);\\\\n\\\\nexport function emojiFn() {\\\\n  return (\\\\n    <Button onClick={action(\\\\\\"clicked\\\\\\")}>\\\\n      <span role=\\\\\\"img\\\\\\" aria-label=\\\\\\"so cool\\\\\\">\\\\n        😀 😎 👍 💯\\\\n      </span>\\\\n    </Button>\\\\n  )\\\\n};\\\\n\\",\\"locationsMap\\":{\\"text\\":{\\"startLoc\\":{\\"col\\":20,\\"line\\":11},\\"endLoc\\":{\\"col\\":1,\\"line\\":13},\\"startBody\\":{\\"col\\":20,\\"line\\":11},\\"endBody\\":{\\"col\\":1,\\"line\\":13}},\\"emoji\\":{\\"startLoc\\":{\\"col\\":21,\\"line\\":15},\\"endLoc\\":{\\"col\\":1,\\"line\\":21},\\"startBody\\":{\\"col\\":21,\\"line\\":15},\\"endBody\\":{\\"col\\":1,\\"line\\":21}},\\"emoji-fn\\":{\\"startLoc\\":{\\"col\\":7,\\"line\\":23},\\"endLoc\\":{\\"col\\":1,\\"line\\":31},\\"startBody\\":{\\"col\\":7,\\"line\\":23},\\"endBody\\":{\\"col\\":1,\\"line\\":31}}}},},
  title: \\"Button\\",
  excludeStories: [\\"text\\"],
  includeStories: /emoji.*/
};

export const text = () => (
  <Button onClick={action(\\"clicked\\")}>Hello Button</Button>
);

export const emoji = () => (
  <Button onClick={action(\\"clicked\\")}>
    <span role=\\"img\\" aria-label=\\"so cool\\">
      😀 😎 👍 💯
    </span>
  </Button>
);;

export const emojiFn = function emojiFn() {
  return (
    <Button onClick={action(\\"clicked\\")}>
      <span role=\\"img\\" aria-label=\\"so cool\\">
        😀 😎 👍 💯
      </span>
    </Button>
  )
};
"
`;
